### PR TITLE
HHH-12763 - Log which JtaPlatform implementation is used at startup on info level 

### PR DESCRIPTION
**Pull request by vladmihalcea**
https://hibernate.atlassian.net/browse/HHH-12763

### CODE REVIEW

**Comment by vladmihalcea** [comment]
@aepedraza I reworked your PR and changed it to use the Core Logger messages. Also, I wanted to print the implementation class only and not force to retrieve the TM or UT since the Service has just been created, but not initialized.

If anyone agrees, I'll close the other PR and integrate this one.

File: hibernate-core/src/main/java/org/hibernate/engine/transaction/jta/platform/internal/JtaPlatformInitiator.java [hunk 1]
```diff
@@ -49,6 +49,7 @@ public JtaPlatform initiateService(Map configurationValues, ServiceRegistryImple
 			platform = getFallbackProvider( configurationValues, registry );
 		}
 
+		LOG.usingJtaPlatform( platform != null ? platform.getClass().getName() : "null" );
```
Comment (gsmet) [inline hibernate-core/src/main/java/org/hibernate/engine/transaction/jta/platform/internal/JtaPlatformInitiator.java]: IMHO let's not log anything if null.

File: hibernate-core/src/test/java/org/hibernate/test/tool/schema/JtaPlatformLoggingTest.java [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+	@Override
+	protected void addSettings(Map settings) {
+		TestingJtaBootstrap.prepare( settings );
+		settings.put( AvailableSettings.TRANSACTION_COORDINATOR_STRATEGY, "jta" );
+	}
+
+	@Test
+	public void test() {
+		assertEquals(
+				"HHH000485: Illegally attempted to associate a proxy for entity [org.hibernate.session.AssociateEntityWithTwoSessionsTest$Location] with id [1] with two open sessions.",
```
Comment (gsmet) [inline hibernate-core/src/test/java/org/hibernate/test/tool/schema/JtaPlatformLoggingTest.java]: This is weird? Why is it about a totally different message?

**Comment by vladmihalcea** [comment]
@gsmet I'll add it for 5.4. The assert was fixed. Now, check out my comment about the logging `null`. If we agree on that, I'll integrate it.

File: hibernate-core/src/main/java/org/hibernate/engine/transaction/jta/platform/internal/JtaPlatformInitiator.java (hunk 1 above)
Comment (gsmet) [reply hibernate-core/src/main/java/org/hibernate/engine/transaction/jta/platform/internal/JtaPlatformInitiator.java]: OK, makes sense. Let's wait for the Travis build to pass before merging to master.

**Comment by aepedraza** [comment]
Hi @vladmihalcea. Thanks for your review. Now I know how to add and test logging. BTW: _"add it to 5.4"_ does this mean to add it to a specific branch or something else?

**Comment by gsmet** [comment]
Merged, thanks to both of you!

**Pull request closed without merging**